README.md

```diff
@@ -1,13 +1,100 @@
 # ember-css-composer
 
-This README outlines the details of collaborating on this Ember addon.
+This addon provides a few simple tools to move the CSS-dependent configuration of components out of 
+Javascript/Handlebars and moves it into a CSS-like configuration file. There are two things this
+might be good for:
+
+1. If you are developer of an addon that exposes components and you want the consumers of your addon to 
+easily override the different class names for the various states of your components (rather than exposing
+configuration options on the component itself)
+
+2. If you're using Functional CSS and you want a way to compose your classes together into named components
+so you don't have needlessly long lists of class names everywhere. 
+
+## Examples
+
+### A List Component
+
+Given this set of files:
+
+```js
+// addon/components/a-list.js
+import Component from 'ember-component';
+
+export default Component.extend({
+  classNames: ['@a-list-component'],
+  classNameBindings: ['isOpen:@a-list-is-open:@a-list-is-closed'],
+  isOpen: false
+});
+```
+
+```hbs
+{{! addon/templates/components/a-list.hbs }}
+{{#each list as |listItem index|}}
+  <li class="@a-list-item">
+    {{yield listItem index}}
+  </li>
+{{/each}}
+```
+
+```css
+/* app/styles/app.css-compose */
+@composer {
+  @a-list-component {
+    composes: 'my-list', 'background-g3', 'font-size-big';
+  }
+
+  @a-list-is-open {
+    composes: 'my-list__open', 'background-g6';
+  }
+
+  @a-list-is-closed {
+    composes: 'my-list__closed';
+  }
+
+  @a-list-item {
+    composes: 'my-list-item', 'color-g4-hover';
+  }
+}
+```
+
+Given all of these settings, this will transpile those files into something like this:
+
+```js
+// addon/components/a-list.js
+import Component from 'ember-component';
+import _classifyMacro from 'ember-css-composer/classify-macro';
+
+export default Component.extend({
+  classNames: ['my-list', 'background-g3', 'font-size-big'],
+  classNameBindings: ['_isOpen'],
+  _isOpen: _classNameMacro('isOpen', ['my-list__open', 'background-g6'], ['my-list__closed']),
```

Comment:
This is not _actually_ what happens...this line would look more like:

```js
_classNameMacro('isOpen', _classify('a-list-is-open'), _classify('a-list-is-closed'))
```

...but, I wanted to demonstrate in the README that those functions will expand to those class names. 

Also — in a _future_ version of this — we could remove the `_classify` function altogether by evaluating that function at build time.